### PR TITLE
NO-SNOW: improve README and Snowpipe Streaming Example

### PR DESCRIPTION
**Pull request by sfc-gh-tzhang**
Improve README and Snowpipe Streaming Example, it should be more clear for customers now

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +53,30 @@ dependencies {
...
+    1. [Here](https://docs.snowflake.com/en/user-guide/key-pair-auth.html#configuring-key-pair-authentication) are the steps required to generate a private key.
+2. Creates a `SnowflakeStreamingIngestClient` which can be used to open one or more Streaming Channels against a table.
+3. Creates a `SnowflakeStreamingIngestChannel` against a Database, Schema and Table.
+    1. Please note: The database, schema and table is expected to be presented before opening the Channel. Example SQL queries to create them
+```sql
+create or replace database MY_DATABASE;
+create or replace schema MY_SCHEMA;
+create or replace table MY_TABLE(c1 number);
+```
+4. Inserts a few rows (1000) into the channel created in 3rd step using the `insertRows` API on the Channel object
```
Comment (sfc-gh-mnaides) [inline README.md:75]: "Inserts 1000 rows into the channel"

File: src/main/java/net/snowflake/ingest/streaming/example/SnowflakeStreamingIngestExample.java [hunk 2]
```diff
@@ -38,56 +41,59 @@ public static void main(String[] args) throws Exception {
 
     // Create a streaming ingest client
     try (SnowflakeStreamingIngestClient client =
-        SnowflakeStreamingIngestClientFactory.builder("CLIENT").setProperties(props).build()) {
+        SnowflakeStreamingIngestClientFactory.builder("MY_CLIENT").setProperties(props).build()) {
 
-      // Create an open channel request on table T_STREAMINGINGEST
+      // Create an open channel request on table MY_TABLE, note that the corresponding
+      // db/schema/table needs to be presented
```
Comment (sfc-gh-mnaides) [inline src/main/java/net/snowflake/ingest/streaming/example/SnowflakeStreamingIngestExample.java:47]: "to be present"

File: src/main/java/net/snowflake/ingest/streaming/example/SnowflakeStreamingIngestExample.java [hunk 3]
```diff
@@ -38,56 +41,59 @@ public static void main(String[] args) throws Exception {
...
+              .setOnErrorOption(
+                  OpenChannelRequest.OnErrorOption.CONTINUE) // Another ON_ERROR option is ABORT
               .build();
 
       // Open a streaming ingest channel from the given client
       SnowflakeStreamingIngestChannel channel1 = client.openChannel(request1);
 
-      // Insert few rows into the channel (Using insertRows API), along with the offset Token which
-      // corresponds to the row number
+      // Insert few rows into the channel (Using insertRows API)
```
Comment (sfc-gh-mnaides) [inline src/main/java/net/snowflake/ingest/streaming/example/SnowflakeStreamingIngestExample.java:61]: "Insert rows into ..."

**Review by sfc-gh-mnaides — APPROVED** [review]
Some minor copy suggestions, otherwise LGTM

File: README.md [hunk 4]
```diff
@@ -52,6 +53,30 @@ dependencies {
...
+## Snowpipe Streaming
+
+Check out `SnowflakeStreamingIngestExample.java`, which performs following operations:
+1. Reads a JSON file which contains details regarding Snowflake Account, User, Role and Private Key. Take a look at `profile_streaming.json.example` for more details.
+    1. [Here](https://docs.snowflake.com/en/user-guide/key-pair-auth.html#configuring-key-pair-authentication) are the steps required to generate a private key.
+2. Creates a `SnowflakeStreamingIngestClient` which can be used to open one or more Streaming Channels against a table.
+3. Creates a `SnowflakeStreamingIngestChannel` against a Database, Schema and Table.
+    1. Please note: The database, schema and table is expected to be presented before opening the Channel. Example SQL queries to create them
+```sql
+create or replace database MY_DATABASE;
```
[the remark this answers is not in the document]
Comment (sfc-gh-tzhang) [reply README.md:71]: I find this useful as it shows you the db/schema/table names need to match when you create the channel, as well as the column name when you do insertRows

**Pull request merged**